registries/deep_rag.hocon

```diff
@@ -0,0 +1,180 @@
+""",
+            "tools": ["rough_structure", "create_network"],
+
+            "allow": {
+                "to_upstream": {
+                    # Specifying this allows sly_data keys from this network to be passed back
+                    # to the calling client
+                    "sly_data": ["agent_reservations", "aa_grouping_json"]
+                }
+            }
+        },
+
+        {
+            "name": "rough_structure",
+            "function": {
+                "description": """
+Creates the grouping_json
+""",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
+                        "file_list": {
+                            "type": "array",
+                            "description": """
+A list of file names that together contains the content to explore.
+The order of these file names can matter.
+""",
+                            "items": {
+                                "type": "string"
+                            }
+                        },
+                        "user_description": {
+                            "type": "string",
+                            "description": """
+This is a description of how the individual files in the file_list might be organized.
+"""
+                        },
+                        "grouping_constraints": {
+                            "type": "string",
+                            "description": """
+This is an optional description of any grouping constraints from the user.
+"""
+                        },
+                    }
+                },
+            },
+            "instructions": """
+Examine the names of the files in the file_list and take into account any user_description
+of the file naming structure to come up with a JSON structure
+which describes the following:
+* A short "name" for the body of work that is being pieced up which must be less that 40 characters.
+* A "description" of the overall strategy for how the files are to be grouped.
+    If you must mention the word 'files' in your description, use the word 'content' instead.
+* A "groups" list which contains one JSON entry for each of the groups, each entry of which has:
+    1. A short "name" for the group which is unique, less than 40 characters and
+        is the primary reference to the group in snake case.
+    2. A "description" of the group and how it is unique from the other groups.
+    3. A "files" dictionary whose keys are each of the files that belong in the group and
+       whose values are a snake case translation of the file name.
+
+Constraints:
+* Each and every file in the file_list must be represented in
+  at least one group.
+* Any single group may have at least one, but never more than 7 files in it.
+* There can be at most 7 groups.
+* Whenever possible follow the advice in the grouping_constraints, if that is present.
+
+Output the JSON structure described.
+""",
```

Comment:
One text-heavy agent to create the structure that describes how the files will be grouped.